add TLS/SSL options for connecting to MQTT server

andreasbrett opened the pull request:
Quick implementation of what @Diaoul requested in https://github.com/Koenkk/zigbee2mqtt/issues/1700.

andreasbrett committed:
add TLS/SSL options for connecting to MQTT server

Diaoul commented on lib/mqtt.js:
```diff
@@ -20,6 +20,16 @@ class MQTT {
+            logger.debug(` - CA cert path = ${mqttSettings.ca}`);
+            options.key = mqttSettings.key;
+            options.cert = mqttSettings.cert;
+            options.ca = mqttSettings.ca;
```
Don't you need to read the file like in [the example](https://github.com/mqttjs/MQTT.js/blob/master/examples/tls%20client/mqttclient.js#L20) such that `options.key` and friends contain the *contents* of the files and not the path? Or is that done in a previous step?

andreasbrett replied:
You're absolutely right. Thought mqtt.js would load them but thanks for your hint. Fixed now.

andreasbrett committed:
Update mqtt.js

andreasbrett committed:
gotta love travis

andreasbrett commented:
@Koenkk Since @Diaoul will only be able to test in a couple of weeks I tested it myself. Good opportunity to up the security of my MQTT server. Hardest thing was to re-configure MQTT but once that was done connecting via z2m worked right away.

PR can be merged!

andreasbrett committed:
Update mqtt.js

separated processing of mqtt.ca and mqtt.key/mqtt.cert in order to allow for "server-authentication only" as well as "server+client authentication"

andreasbrett committed:
Update mqtt.js

Koenkk commented:
Thanks!
